src/test/java/org/kitteh/irc/client/library/util/MaskTest.java

```diff
@@ -0,0 +1,114 @@
+package org.kitteh.irc.client.library.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.kitteh.irc.client.library.element.Channel;
+import org.kitteh.irc.client.library.element.User;
+import org.kitteh.irc.client.library.test.TestUser;
+import org.mockito.Mockito;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * It's a mask test.
+ */
+public class MaskTest {
```

Comment:
add a test for spoofed masks? `freenode/staff/potato` comes to mind and `lol` (invalid hostname), or `catbot is catbot@4,1nyan7nyan8nyan3nyan11nyan10nyan12nyan13nyan * nyan catbot`